www/src/main/java/SiteBuilder.java

```diff
@@ -0,0 +1,266 @@
+import static utils.Utils.getMarkdownCodeBlockWithCodeType;
+import static utils.Utils.getPostFrontMatter;
+import static utils.Utils.removeLicenceHeader;
+import static utils.Utils.renameAndMoveFile;
+
+/**
+ * Main class of the site creator project.
+ */
+public class SiteBuilder {
+    // Setup logger.
+    private static final Logger logger = Logger.getLogger(SiteBuilder.class.getName());
+
+    public static void main(String[] args) {
+        try {
+            // First delete already created posts.
+            deleteDirectory(GUIDES_DIR);
+            deleteDirectory(INTEGRATION_TUTORIALS_DIR);
+            deleteNonIndexFiles(new File(INTRO_DIR));
+            // Create needed directory structure.
+            createDirectory(TEMP_DIR);
+            createDirectory(_GUIDE_TEMPLATES_DIR);
+            // get a copy of examples directory.
+            copyDirectoryContent(REPO_EXAMPLES_DIR, TEMP_DIR);
+            // Process repository to generate guide templates.
+            processDirectory(TEMP_DIR);
+            // Copy tempDirectory content to hugo content directory.
+            copyDirectoryContent(TEMP_DIR, _GUIDE_TEMPLATES_DIR);
+        } catch (ServiceException e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        } finally {
+            deleteDirectory(TEMP_DIR);
+        }
+    }
+
+    /**
+     * Process files inside given directory.
+     *
+     * @param directoryPath path of the directory
+     */
+    private static void processDirectory(String directoryPath) {
+        File folder = new File(directoryPath);
+        File[] listOfFiles = folder.listFiles();
+
+        if (listOfFiles != null) {
+            for (File file : listOfFiles) {
+                if (file.isFile() && (file.getName().equals(README_MD))) {
+                    processReadmeFile(file);
+                    renameReadmeFile(file);
+                } else if (file.getName().equals(INDEX_MD)) {
+                    processReadmeFile(file);
+                } else if (file.getName().equals(INTRO_MD)) {
+                    processReadmeFile(file);
+                    renameAndMoveFile(file, CONTENT_INTRO_DIR, getCurrentDirectoryName(file.getParent()));
+                } else if (file.isDirectory()) {
+                    processDirectory(file.getPath());
+
+                }
+            }
+        }
+    }
+
+    /**
+     * Process a given README.md by reading through lines.
+     *
+     * @param file README.md file
+     */
+    private static void processReadmeFile(File file) {
+        try {
+            String readMeFileContent = IOUtils
+                    .toString(new FileInputStream(file), String.valueOf(StandardCharsets.UTF_8));
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line;
+            int lineNumber = 0;
+            int frontMatterSignNo = 0;
+            boolean isIntroMdFile = file.getName().equals(INTRO_MD);
+
```

Comment:
Shall we remove `frontMatterSignNo`  and `isIntroMdFile` variables? They are not used within this method.

Reply:
Done